src/pages/index.js

```diff
@@ -170,16 +242,187 @@ profileCard.setEventListeners();
+  .catch((err) => {
+    console.error("An error was found", err);
+  });
+
+/* -------------------------------------------------------------------------- */
+/*                                fetch request                               */
+/* -------------------------------------------------------------------------- */
+api
+  .userProfileInfo()
+  .then((res) => res.json())
+  .then((info) => {
+    console.log(info);
+    userInfo.setUserInfo({
+      name: info.name,
+      job: info.about,
+    });
+    console.log(userInfo);
+  })
+  .catch((err) => {
+    console.error("an error has occurred", err);
+  })
+  .finally(() => {
+    console.log("done");
+  });
+
+/* -------------------------------------------------------------------------- */
+/*                                card requests                               */
+/* -------------------------------------------------------------------------- */
+
+function handleLikeButton(card) {
+  if (card._isLiked) {
+    api
+      .removeLikeButtonRequest(card._id)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.error("an error has occurred", err);
+      })
+      .finally(() => {
+        console.log("done");
+      });
+  } else {
+    api
+      .likeButtonRequest(card._id)
+      .then((res) => {
+        card.updateLikes(res.isLiked);
+      })
+      .catch((err) => {
+        console.error("an error has occurred", err);
+      })
+      .finally(() => {
+        console.log("done");
+      });
+  }
+}
+
+/* -------------------------------------------------------------------------- */
+/*                            profile image request                            */
+/* -------------------------------------------------------------------------- */
+
+function setProfilePicture(linkInput) {
+  profileImage.src = linkInput;
+}
+
+const updateProfilePicture = new PopupWithForm({
+  popupSelector: "#change-avatar-modal",
+  handleFormSubmit: (inputValues) => {
+    const linkInput = inputValues.link;
+    console.log(linkInput);
+    api
+      .updateProfilePictureRequest({ linkInput: linkInput })
+      .then((res) => {
+        console.log(res);
+        setProfilePicture(linkInput);
+        updateProfilePicture.close();
+        // this is where the pushing of the link goes through to the src on the img
+      })
+      .catch((err) => {
+        console.error("an error has occurred", err);
+      })
+      .finally(() => {
+        console.log("done");
+      });
+  },
+});
+
+profileImageContainer.addEventListener("mousedown", () => {
+  updateProfilePicture.open();
+});
+
+updateProfilePicture.updateProfileImageEventListeners();
+
+updateProfilePicture.setEventListeners();
+
+api
+  .userProfileInfo()
+  .then((res) => res.json())
+  .then((info) => {
+    const avatarImage = info.avatar;
+    console.log(avatarImage);
+    setProfilePicture(avatarImage);
+  });
```

Comment:
Please, delete an extra `api.userProfileInfo()` from `index.js`. You've requested the data previously